libsolidity/parsing/Scanner.cpp

```diff
@@ -755,8 +772,18 @@ Token::Value Scanner::scanNumber(char _charSeen)
 				addLiteralCharAndAdvance();
 				if (!isHexDigit(m_char))
 					return Token::Illegal; // we must have at least one hex digit after 'x'/'X'
-				while (isHexDigit(m_char))
+				char last = m_char;
+				while (isHexDigit(m_char) || m_char == '_') // Unlike decimal digits, we keep the underscores for later validation
+				{
+					if (m_char == '_') {
+						if (last == '_')
```

Comment:
This will always be triggered for a single `_`.

Reply:
@chriseth  I didn't understand what you meant.

The idea is this: Line 773 takes invalidates leading underscore. After that, we only need to avoid double . So, we store the last character. If current char is '' and last char was also '_' then it is invalid. Otherwise we can add the current character and validate later in SyntaxChecker.

Reply:
Ah right, now I see it, sorry. If you want, you can contract the two if statements. Otherwise, at least move the `{` to the next line :-)

Reply:
@chriseth Done. Too many formatting issues, sorry :( .